plugins/woocommerce/tests/e2e-pw/tests/merchant/product-reviews.spec.js

```diff
@@ -196,6 +196,109 @@ test.describe(
 			).toBeVisible();
 		} );
 
+		test( 'can approve a product review', async ( { page, reviews } ) => {
+			const review = reviews[ 0 ]; // Select the first review for approval
+
+			// Go to the Product Reviews page
+			await page.goto(
+				`wp-admin/edit.php?post_type=product&page=product-reviews`
+			);
+
+			// Locate the review to be approved
+			const reviewRow = page.locator( `#comment-${ review.id }` );
+
+			// Ensure the review is not already approved
+			const approveButton = reviewRow.getByRole( 'button', {
+				name: 'Approve',
+			} );
+			const isApproveButtonVisible = await approveButton.isVisible();
+
+			if ( isApproveButtonVisible ) {
+				// Scroll to the review row and hover to reveal action buttons
+				await reviewRow.scrollIntoViewIfNeeded();
+				await reviewRow.hover();
+
+				// Click the "Approve" button and wait for the state to become 'networkidle'
+				await approveButton.click();
+				await page.waitForLoadState( 'networkidle' );
+
+				// Verify the review is approved by checking the presence of "Unapprove" button
+				const unapproveButton = reviewRow.getByRole( 'button', {
+					name: 'Unapprove',
+				} );
+				await expect( unapproveButton ).toBeVisible;
+			}
+		} );
+
+		test( 'can mark a product review as spam', async ( {
+			page,
+			reviews,
+		} ) => {
+			const review = reviews[ 0 ]; // Select the first review to mark as spam
+
+			// Go to the Product Reviews page
+			await page.goto(
+				`wp-admin/edit.php?post_type=product&page=product-reviews`
+			);
+
+			// Locate the review to be marked as spam
+			const reviewRow = page.locator( `#comment-${ review.id }` );
+			await reviewRow.hover();
+
+			// Select Spam action and verify the review is marked as spam
+			await reviewRow.getByRole( 'button', { name: 'Spam' } ).click();
+
+			// Navigate to the "Spam" tab to verify the review is marked as spam
+			await page.click( 'a[href*="comment_status=spam"]' );
+
+			// Check if the spammed review row is located and verify the review is
+			// marked as spam by confirming "Not Spam" button is visible
+			const spammedReviewRow = page.locator( `#comment-${ review.id }` );
+			await spammedReviewRow.isVisible();
+			await reviewRow.hover();
```

Comment:
You are hovering the `reviewRow` instead of `spammedReviewRow`. It works, because they have the same locator.

Reply:
Right, updated now. Thank you very much!